src/app/public/src/modules/link/link.directive.ts

```diff
@@ -0,0 +1,31 @@
+import { Directive, ElementRef, HostListener, Input, Renderer2 } from '@angular/core';
+
+import { StacheNavService } from '../nav';
+
+@Directive({
+  selector: '[stacheRouterLink]'
+})
+export class StacheRouterLinkDirective {
+  @Input()
+  public stacheRouterLink: string;
+
+  @Input()
+  public fragment: string;
+
+  constructor(
+    private navService: StacheNavService,
+    private el: ElementRef,
+    private renderer: Renderer2
+  ) {
+    this.renderer.setStyle(this.el.nativeElement, 'cursor', 'pointer');
+   }
+
+  @HostListener('click', ['$event'])
+  public navigate(event: any): void {
+    event.preventDefault();
+    this.navService.navigate({
+      path: this.stacheRouterLink,
+      fragment: this.fragment ? this.fragment : undefined
```

Comment:
The logic on this line could simply be:
`fragment: this.fragment`

Reply:
I think I was doing something different with this logic initially. Good catch, I'll remove it.